### ISSUE TITLE
Define CMake target instead of running CMinx during configuration stage

### ISSUE THREAD

**Issue by peanutfun** [CMakePP/CMinx#108]
**Is your feature request related to a problem? Please describe.**
CMinx currently offers two types of installation: One via `pip`, which effectively only installs the `cminx` script, and one via CMake, which also installs the CMake modules. If CMinx is used in a downstream project, one can use the `cminx_gen_rst` function from the installed CMake module to generate the CMake documentation for said project. However, this function works at "configure time", i.e., when CMake is executed. I think this works agains the notion of a "build". Building a documentation is a typical task once configuration of a project is complete. In particular, it's only done on request by the user, to keep configuration as quick as possible.

**Describe the solution you'd like**
I propose that the function `cminx_gen_rst` does not run the `cminx` command immediately but uses `add_custom_command` or `add_custom_target` to define a CMake target that then runs `cminx`. After configuring the project, users can then decide to build the "CMinx target", and are able to use this target as dependency for their typical `doc` targets.

**Describe alternatives you've considered**
None.

**Additional context**
None.

**Comment by ryanmrichard** [comment]
Looking at this a bit more I'd have to break the `cminx_gen_rst` API to add an option for toggling a target. So I don't want to go that route. I think a better option is to write a new function which relies on a target. Users can then use the function they prefer.

That said, the CMakePP team's cycles are spread a bit thin at the moment, so I don't want to prioritize this issue unless it's affecting users. So, @peanutfun is this issue purely academic, or is this something that is (or will be) affecting your project(s)?

**Comment by peanutfun** [comment]
No, this does not affect my projects, it's mainly a suggestion. Feel free to give this issue a low priority.
